src/components/Modal/Modal.tsx

```diff
@@ -70,32 +113,75 @@ export const BaseModal: ChakraComponent<
       headingSize: "heading4",
     });
 
-    return (
-      <ChakraModal
-        id={id}
-        isOpen={isOpen}
-        onClose={onClose}
-        scrollBehavior="inside"
-        size={size}
-        {...rest}
-      >
-        <ModalOverlay />
-        <ModalContent>
-          <ModalHeader>{finalTitle}</ModalHeader>
-          <ModalCloseButton />
-          <ModalBody>{bodyContent}</ModalBody>
+    if (
+      (!isDefaultType(type) && onClose) ||
+      (isDefaultType(type) && (onCancel || onConfirm))
+    ) {
+      console.warn(
+        "NYPL Reservoir Modal: The `onClose`, `onConfirm`, and `onCancel` props have all been set, but they can not all be used together. Only `onClose` will be used, while `onConfirm` and `onCancel` will be ignored."
+      );
+    }
 
-          <ModalFooter>
-            <ButtonGroup>
-              <Button id="modal-close-btn" onClick={onClose}>
-                {closeButtonLabel}
-              </Button>
-            </ButtonGroup>
-          </ModalFooter>
-        </ModalContent>
-      </ChakraModal>
+    return (
+      <>
+        {isDefaultType(type) && (
+          <ChakraModal
+            id={id}
+            isOpen={isOpen}
+            onClose={onClose}
+            scrollBehavior="inside"
+            size={size}
+            {...rest}
+          >
+            <ModalOverlay />
+            <ModalContent>
+              <ModalHeader>{finalTitle}</ModalHeader>
+              <ModalCloseButton />
+              <ModalBody>{bodyContent}</ModalBody>
+              <ModalFooter>
+                <ButtonGroup>
+                  <Button id="modal-close-btn" onClick={onClose}>
+                    {closeButtonLabel}
+                  </Button>
+                </ButtonGroup>
+              </ModalFooter>
+            </ModalContent>
+          </ChakraModal>
+        )}
+        {!isDefaultType(type) && (
```

Comment:
I know this is using the `TagSet` pattern but I wonder why we didn't use a tertiary there. Do you think this could be easier to read if it was used here?

Reply:
yes